docs/Developers/Newcomers_Guide.rst

```diff
@@ -239,9 +239,22 @@ where ``myfork`` is the name of your fork, and ``fork_link`` is a link to your
 fork repository.
 
 .. note::
-   It is important that you do not make your changes on the master branch. To
-   start working on an issue, you first need to create a new branch where you
-   will work.
+   It is important that you **DO NOT** make your changes on the master branch
+   for the following reasons:
+
+   - master should always be considered as the release version of the product.
+   - If you push an unfinished or untested state of commits to remote, there
+     is a possibility that master ends up in a broken state.
+   - Other developers may start working on a potentially broken master, which
```

Comment:
Some of these "reasons" are about the master branch on the central repository, whereas this issue is about pushing to `master` of a personal fork, and that branch being used on a *pull request* to the central repository.

The reasons to not use `<my fork>:master` in pull requests center around the fact that rebasing on top of `<origin>:master` is quite a bit harder to explain.

Also, if `<my fork>:master` is used in a pull request, `master` needs to keep having that patch on it until the patch is merged.  If the developer wants to work on a second change, they will either

1. use `master` again and the first PR now has the commit for the second issue.
2. create a feature branch, so branching is unavoidable if the developer intends to work on more than one issue concurrently.